### PR TITLE
Restore update_manifest.py to fix failing "Prepare release asset" Git…

### LINKED ISSUE #260: Unable to update to version 2.3.0
I'm unable to update from version 2.2.1 to the latest 2.3.0 release. Looking at the logs, it complains that the file hass-simpleicons.zip is missing and indeed this release does not include it. @vigonotion can you please check? Thank you!

HA Logs
```
2024-11-18 23:40:24.275 INFO (MainThread) [custom_components.hacs] Starting download, v2.3.0
2024-11-18 23:40:24.284 INFO (MainThread) [custom_components.hacs] <Integration vigonotion/hass-simpleicons> Running pre installation steps
2024-11-18 23:40:24.284 INFO (MainThread) [custom_components.hacs] <Integration vigonotion/hass-simpleicons> Pre installation steps completed
2024-11-18 23:40:24.286 INFO (MainThread) [custom_components.hacs] <Integration vigonotion/hass-simpleicons> Running installation steps
2024-11-18 23:40:27.099 ERROR (MainThread) [custom_components.hacs] Download failed - Got status code 404 when trying to download https://github.com/vigonotion/hass-simpleicons/releases/download/v2.3.0/hass-simpleicons.zip
Traceback (most recent call last):
  File "/config/custom_components/hacs/base.py", line 709, in async_download_file
    raise HacsException(
custom_components.hacs.exceptions.HacsException: Got status code 404 when trying to download https://github.com/vigonotion/hass-simpleicons/releases/download/v2.3.0/hass-simpleicons.zip
2024-11-18 23:40:27.116 ERROR (MainThread) [custom_components.hacs] <Integration vigonotion/hass-simpleicons> Failed to download https://github.com/vigonotion/hass-simpleicons/releases/download/tags/v2.3.0/hass-simpleicons.zip
2024-11-18 23:40:27.140 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [547656952000] Unexpected exception
Traceback (most recent call last):
  File "/config/custom_components/hacs/repositories/base.py", line 1436, in async_download_repository
    await self.async_install(version=ref)
  File "/config/custom_components/hacs/repositories/base.py", line 884, in async_install
    await self.async_install_repository(version=version)
  File "/config/custom_components/hacs/repositories/base.py", line 986, in async_install_repository
    raise HacsException("Could not download, see log for details")
custom_components.hacs.exceptions.HacsException: Could not download, see log for details

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/hacs/update.py", line 88, in async_install
    await self.repository.async_download_repository(ref=version or self.latest_version)
  File "/config/custom_components/hacs/repositories/base.py", line 1438, in async_download_repository
    raise HacsException(
custom_components.hacs.exceptions.HacsException: Downloading vigonotion/hass-simpleicons with version v2.3.0 failed with (Could not download, see log for details)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 241, in handle_call_service
    response = await hass.services.async_call(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2763, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2806, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 999, in entity_service_call
    single_response = await _handle_entity_call(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1071, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/update/__init__.py", line 154, in async_install
    await entity.async_install_with_progress(version, backup)
  File "/usr/src/homeassistant/homeassistant/components/update/__init__.py", line 458, in async_install_with_progress
    await self.async_install(version, backup)
  File "/config/custom_components/hacs/update.py", line 90, in async_install
    raise HomeAssistantError(exception) from exception
homeassistant.exceptions.HomeAssistantError: Downloading vigonotion/hass-simpleicons with version v2.3.0 failed with (Could not download, see log for details)
```

### PR DESCRIPTION
**Pull request by akston**
…Hub Action

fixes #260

### CODE REVIEW

File: manage/update_manifest.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        if value in ["--version", "-V"]:
+            version = sys.argv[index + 1]
+    with open(f"{os.getcwd()}/custom_components/simpleicons/manifest.json") as manifestfile:
+        manifest = json.load(manifestfile)
+    manifest["version"] = version
+    with open(
+        f"{os.getcwd()}/custom_components/simpleicons/manifest.json", "w"
+    ) as manifestfile:
+        manifestfile.write(json.dumps(manifest, indent=4, sort_keys=True))
+update_manifest()
```
Comment (TurnrDev) [inline manage/update_manifest.py:18]:
```suggestion
"""Update the manifest file."""

import sys
import json
import os

def update_manifest():
    """Update the manifest file."""
    
    version = "0.0.0"
    
    for index, value in enumerate(sys.argv):
        if value in ["--version", "-V"]:
            version = sys.argv[index + 1]
            
    with open(f"{os.getcwd()}/custom_components/simpleicons/manifest.json") as manifestfile:
        manifest = json.load(manifestfile)
    manifest["version"] = version
    
    with open(
        f"{os.getcwd()}/custom_components/simpleicons/manifest.json", "w"
    ) as manifestfile:
        manifestfile.write(json.dumps(manifest, indent=4, sort_keys=True))

update_manifest()
```

**Review by TurnrDev — COMMENTED** [review]
Let's make this easier to read, eh?

File: manage/update_manifest.py (hunk 1 above)
Comment (akston) [reply manage/update_manifest.py:18]: This is not my code, rather a partial revert of 0904e22. I don't disagree with the suggested change, but my goal is simply to restore functionality in HACS. I'll let @vigonotion decide on any optimizations.

Comment (akston) [reply manage/update_manifest.py]: My apologies, I didn't review the change very well, and it appears my copy/paste into VS Code didn't preserve formatting for some reason. I've cleaned it up. 👍

**Comment by vigonotion** [comment]
thanks for fixing this!

**Pull request merged**